frontend/public/components/modals/configure-cluster-upstream-modal.tsx

```diff
@@ -0,0 +1,118 @@
+    const [invalidCustomUrl, setInvalidCustomUrl] = React.useState(false);
+
+    const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+      e.preventDefault();
+      if (selectedRadio === 'custom') {
+        if (!customUrl) {
+          setInvalidCustomUrl(true);
+          return;
+        } else if (customUrl === currentUpstream) {
+          return handlePromise(Promise.resolve(), close);
+        }
+      } else if (!currentUpstream) {
+        return handlePromise(Promise.resolve(), close);
+      }
+      const value = selectedRadio === 'default' ? null : customUrl;
+      const patch = [{ op: 'add', path: '/spec/upstream', value }];
+      return handlePromise(k8sPatch(ClusterVersionModel, cv, patch), close);
+    };
+    const { t } = useTranslation();
+
+    return (
+      <form onSubmit={submit} name="form" className="modal-content ">
+        <ModalTitle>{t('public~Edit upstream configuration')}</ModalTitle>
+        <ModalBody>
+          <p>
+            {t(
+              'public~Select a configuration to receive updates. Updates can be configured to receive information from Red Hat or a custom update service.',
+            )}
+          </p>
+          <ExternalLink
+            href={'/'} //TODO
+            text={t('public~Learn more about OpenShift local update services.')}
+          />
+          <RadioInput
+            onChange={(e) => {
+              setSelectedRadio(e.target.value);
+              setInvalidCustomUrl(false);
+            }}
+            value={'default'}
+            checked={selectedRadio === 'default'}
+            title={t('public~Default. Receive update information from Red Hat.')}
+          >
+            <TextInput
+              id={'cluster-version-default-upstream-server-url'}
+              type="url"
+              isReadOnly
+              isDisabled
+              value={CLUSTER_VERSION_DEFAULT_UPSTREAM_SERVER_URL}
+            />
+          </RadioInput>
+          <RadioInput
+            onChange={(e) => setSelectedRadio(e.target.value)}
+            value={'custom'}
+            checked={selectedRadio === 'custom'}
+            title={t('public~Custom update service.')}
+          >
+            <TextInput
+              id={'cluster-version-custom-upstream-server-url'}
+              type="url"
+              placeholder={CLUSTER_VERSION_CUSTOM_UPSTREAM_SERVER_URL_PLACEHOLDER}
+              value={customUrl}
+              onChange={(text) => {
+                setSelectedRadio('custom');
+                setCustomUrl(text);
+                setInvalidCustomUrl(false);
+              }}
+              validated={invalidCustomUrl ? 'error' : 'default'}
+            />
+          </RadioInput>
+        </ModalBody>
+        <ModalSubmitFooter
+          errorMessage={props.errorMessage}
+          inProgress={props.inProgress}
+          submitText={t('public~Save')}
+          cancel={props.cancel}
+        />
```

Comment:
When `Custom update service` input is cleared `Save` button is enabled and clicking `Save` will return error icon but no error message, shall we add error message as well? 
<img width="629" alt="Screen Shot 2021-09-01 at 4 29 09 PM" src="https://user-images.githubusercontent.com/12692381/131639128-e2bc49a0-169b-4db0-92f1-e861c17b097f.png">

Reply:
@yapei done! Can you have another look and confirm you see it:
<img width="631" alt="Screen Shot 2021-09-01 at 1 05 42 PM" src="https://user-images.githubusercontent.com/21317056/131719275-b38eb8b8-e8f2-4011-b272-7ad4462b1f8f.png">